bt-target/client/main.lua

```diff
@@ -60,7 +60,9 @@ function playerTargetEnable()
                                     if option.shouldShow == nil or option.shouldShow() then
                                         for _, job in pairs(option.job) do
                                             if job == "all" or job == PlayerJob.name then
-                                                table.insert(NewOptions, option)
+                                                option2 = ShallowCopy(option)
```

Comment:
Would it not be easier to do
```lua
_option = option
_option.shouldShow = nil
```
instead of adding extra functions to copy

Reply:
If you do this, both tables have the same address. If the function inside shouldShow returns true, and after that you target the zone when the function inside returns false, it'd still show the option.

```lua
shouldShow = function()
	if GetEntityHealth(PlayerPedId()) <= 50 then
		return true 
	else        
		return false
	end
end
```

If the player health is under 50, the option will be shown, but if the player healed and tries to target the zone again, the option will still appear if you do that.

Reply:
I'm talking about your "ShallowCopy" function, seems unnecessary to me, you're dumping the conents of one table in an O(n) algorithm when just asigning it to a temp variable would have the same effect in O(1) time

Reply:
Have you tried assigning it to a temp variable and tested it out ingame? It does not work.

Reply:
What I'm saying is you take
```lua
option2 = ShallowCopy(option)
option2.shouldShow = nil
table.insert(NewOptions, option2)
```
And just do
```
_option = option
_option.shouldShow = nil
table.insert(NewOptions, _option)
```
which would copy the table to a temporary variable and you can remove `shouldShow` from the temporary and use that instead of using another function to clone it

Reply:
If you do _option.shouldShow = nil, option.shouldShow will be nil aswell.

![image](https://user-images.githubusercontent.com/15060451/126310848-0b2b0661-bc63-470e-8ed3-827708826dff.png)

Once the function returns true, it will never check again.

Reply:
Alright then, lua is a bit weird with this then I guess, I'll merge this.